odl/solvers/functional/functional.py

```diff
@@ -0,0 +1,1516 @@
+        if not isinstance(func, Functional):
+            raise TypeError('functional {!r} is no a Functional instance'
+                            ''.format(func))
+        if scalar not in func.range:
+            raise TypeError('the scalar {} is not in the range of the'
+                            'functional {!r}'.format(scalar, func))
+
+        Functional.__init__(self, domain=func.domain, linear=func.is_linear,
+                            smooth=func.is_smooth, concave=func.is_concave,
+                            convex=func.is_convex,
+                            grad_lipschitz=func.grad_lipschitz)
+
+        OperatorSum.__init__(self, left=func,
+                             right=ConstantOperator(vector=scalar,
+                                                    domain=func.domain,
+                                                    range=func.range),
+                             tmp_ran=None, tmp_dom=tmp_dom)
+
+    # TODO: Update this if ConstantOperator is updated.
+    @property
+    def scalar(self):
+        return self.right.vector
+
+    @property
+    def gradient(self):
+        """Gradient operator of the FunctionalScalarSum.
+
+        This is the same as the gradient of the original functional.
+        """
+        return self.left.gradient
+
+    def proximal(self, sigma=1.0):
+        """Proximal operator of the FunctionalScalarSum.
+
+        This is the same as the proximal operator of the original
+        functional.
+
+        Parameters
+        ----------
+        sigma : positive float, optional
+            Regularization parameter of the proximal operator.
+
+        Returns
+        -------
+        out : `Operator`
+            Domain and range equal to domain of functional.
+        """
+        return self.left.proximal(sigma)
+
+    @property
+    def conjugate_functional(self):
+        """Convex conjugate functional of FunctionalScalarSum."""
+        return self.left.conjugate_functional - self.scalar
+
+    def derivative(self, point):
+        """Returns the derivative operator of FunctionalScalarSum.
+
+        Parameters
+        ----------
+        point : `LinearSpaceVector`
+            The point in which the gradient is evaluated.
+
+        Returns
+        -------
+        out : `DerivativeOperator`
+            The linear operator that maps ``x --> <x, grad_f(point)>``.
+        """
+        return self.left.derivative(point)
+
+
+class TranslatedFunctional(Functional):
+    """Implementation of the translated functional.
+
+    Given a functional ``f`` and an element ``translation`` in the domain of
+    ``f``, this corresponds to the functional ``f(. - translation)``.
+    """
+    # TODO: if the orginal functional is linear, this will also be linear if
+    # the "total" translation is zero...
+    # Should we check instance type of func, and if it is TranslatedFunctional
+    # try to combined to only one TranslatedFunctional with a total
+    # translation?
```

Comment:
That's a question of efficiency. If the combined translated guy is evaluated often, it can be worth to combine the two translations, if it's used once (or never) it adds overhead. So the answer to this is highly use case dependent. We could have this optional as a possibility.